tools/behavior_simulator/src/state.rs

```diff
@@ -0,0 +1,280 @@
+                    let orientation = match orientation_mode {
+                        OrientationMode::AlignWithPath => {
+                            if step.norm_squared() < f32::EPSILON {
+                                UnitComplex::identity()
+                            } else {
+                                UnitComplex::from_cos_sin_unchecked(step.x, step.y)
+                            }
+                        }
+                        OrientationMode::Override(orientation) => *orientation,
+                    };
+
+                    *robot_to_field = Isometry2::new(
+                        robot_to_field.translation.vector + robot_to_field.rotation * step,
+                        robot_to_field.rotation.angle()
+                            + orientation.angle().clamp(
+                                -std::f32::consts::FRAC_PI_4 * time_step.as_secs_f32(),
+                                std::f32::consts::FRAC_PI_4 * time_step.as_secs_f32(),
+                            ),
+                    )
+                }
+                MotionCommand::InWalkKick {
+                    head: _,
+                    kick,
+                    kicking_side,
+                } => {
+                    if let Some(ball) = self.ball.as_mut() {
+                        let side = match kicking_side {
+                            Side::Left => 1.0,
+                            Side::Right => -1.0,
+                        };
+
+                        // TODO: Check if ball is even in range
+                        // let kick_location = robot_to_field * ();
+
+                        let strength = 1.0;
+                        let direction = match kick {
+                            KickVariant::Forward => vector![1.0, 0.0],
+                            KickVariant::Turn => vector![0.707, 0.707 * side],
+                            KickVariant::Side => vector![0.0, 1.0 * -side],
+                        };
+                        ball.velocity += *robot_to_field * direction * strength;
+                    }
+                }
+                _ => {}
+            }
+
+            let incoming_messages: Vec<_> = incoming_messages
+                .iter()
+                .filter_map(|(sender, message)| {
+                    (sender != player_number).then_some(IncomingMessage::Spl(*message))
+                })
+                .collect();
+            robot.database.main_outputs.game_controller_state = Some(GameControllerState {
+                game_state: GameState::Playing,
+                game_phase: GamePhase::Normal,
+                kicking_team: Team::Uncertain,
+                last_game_state_change: now,
+                penalties: Default::default(),
+                remaining_amount_of_messages: 1200,
+                set_play: None,
+            });
+            let messages = incoming_messages.iter().collect();
+            let messages = BTreeMap::from_iter(once((now, messages)));
```

Comment:
Not easily, no.
The cycler needs a `BTreeMap<SystemTime, Vec<&IncomingMessage>>`, note the reference inside.
That means we need to create a Vec that stores the references as well as another data structure (I chose another Vec) to store to actual values to be referenced.

Reply:
The only way I see to change this is to store `IncomingMessage` instead of `SplMessage` in the state struct. I tried that and it seems to work,
But then we would allow GameController messages to be sent between robots as well, which doesn't make any sense at all.
I think sticking with the current solution is better. What do you think?